### PR TITLE
add translate API; switch most libraries to automatic dependency resolution

### PR DESCRIPTION
**Pull request by elharo**
@chanseokoh @briandealwis

### CODE REVIEW

**Comment by briandealwis** [comment]
Bunch of test failures like the following:
```
testSelectApiClient(com.google.cloud.tools.eclipse.appengine.libraries.ui.ClientApisLibrariesSelectorGroupTest)  Time elapsed: 0.941 sec  <<< FAILURE!
java.lang.AssertionError: Could not find button for googleapiclient
	at org.junit.Assert.fail(Assert.java:88)
	at com.google.cloud.tools.eclipse.appengine.libraries.ui.ClientApisLibrariesSelectorGroupTest.getButton(ClientApisLibrariesSelectorGroupTest.java:140)
	at com.google.cloud.tools.eclipse.appengine.libraries.ui.ClientApisLibrariesSelectorGroupTest.setUp(ClientApisLibrariesSelectorGroupTest.java:60)
```

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java [hunk 1]
```diff
@@ -200,12 +203,54 @@ public synchronized void resolveDependencies() throws CoreException {
...
+    for (LibraryFile file : dependencies) {
+      MavenCoordinates coordinates = file.getMavenCoordinates();
+      String key = coordinates.getGroupId() + ":" + coordinates.getArtifactId();
+      if (map.containsKey(key)) {
+        MavenCoordinates previousCoordinates = map.get(key);
+        if (newer(coordinates, previousCoordinates)) {
+          map.put(key, coordinates);
+          for (int i = 0; i < result.size(); i++) {
+            if (result.get(i).getMavenCoordinates().equals(previousCoordinates)) {
+              result.set(i, file); // replace
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java]: The `result` list seems a bit complicated.  We seem to assume that there's a 1-1 relationship between `LibraryFile` and `MavenCoordinates` (or at least after running through this list, any `LibraryFile`s with these coordinates will be changed to provide this `file` instance, and so we'll have duplicates).  So can't we just have the `map` be `<String,LibraryFile>` instead?

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java [hunk 2]
```diff
@@ -200,12 +203,54 @@ public synchronized void resolveDependencies() throws CoreException {
...
+        result.add(file);
+      }
+      
+    }
+    return result;
+  }
+
+  private static boolean newer(MavenCoordinates coordinates, MavenCoordinates previousCoordinates) {
+    try {
+      Version version1 = new Version(coordinates.getVersion());
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java]: Given these are `_Maven_Coordinates`, perhaps better to use `org.apache.maven.artifact.versioning.ComparableVersion`?

(Could have `MavenCoordinates` implement `Comparable`?)

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java]: Done.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java (hunk 2 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java]: Done.

**Pull request merged**